docs/architecture.md

```diff
@@ -1,188 +1,264 @@
+
+This diagram shows the cluster when initialization of `policy-server` is complete:
 
 ![policy-server initialized](/img/architecture_sequence_03.png)
 
 ### Making Kubernetes aware of the policy
 
+<!--TODO:Next paragraph. Says 'admission reviews', should that maybe be
+'admission requests'-->
+
 The `policy-server` Pods have a
 [`Readiness Probe`](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/),
-`kubewarden-controller` relies on that to know when the `policy-server` Deployment
-is ready to evaluate admission reviews.
+that `kubewarden-controller` uses to check when
+the `policy-server` Deployment is ready to evaluate admission reviews.
+
+<!--TODO:Aware? Is notified? Notifies? Is 'notifies' technically correct? Messages?-->
 
-Once the `policy-server` Deployment is marked as `Ready`, `kubewarden-controller`
-will make the Kubernetes API server aware of the new policy by creating either a
-`MutatingWebhookConfiguration` or a `ValidatingWebhookConfiguration`
-object.
+Once the readiness probe marks a `policy-server` Deployment as `Ready`,
+the `kubewarden-controller` notifies the Kubernetes API server about the new policy
+by creating either a `MutatingWebhookConfiguration`
+or a `ValidatingWebhookConfiguration` object.
 
-Each policy has its dedicated `MutatingWebhookConfiguration`/`ValidatingWebhookConfiguration`
-which points to the Webhook endpoint served by `policy-server`. The endpoint
-is reachable by the `/validate/<policy ID>` URL mentioned before.
+Each policy has a dedicated
+`MutatingWebhookConfiguration` or `ValidatingWebhookConfiguration`
+which points to the Webhook endpoint served by `policy-server`.
+The endpoint is reachable by the `/validate/<policy ID>` URL.
 
 ![Kubernetes Webhook endpoint configuration](/img/architecture_sequence_04.png)
 
 ### Policy in action
 
-Now that all the plumbing has been done, Kubernetes will start sending the
-relevant Admission Review requests to the right `policy-server` endpoint.
+<!--TODO:I made endpoints plural. I think there are many. Is that correct?-->
+Now that all the necessary plumbing is complete,
+Kubernetes starts sending Admission Review requests to the right `policy-server` endpoints.
 
 ![Policy in action](/img/architecture_sequence_05.png)
 
-`policy-server` receives the Admission Request object and, based on the
-endpoint that received the request, uses the right policy to evaluate it.
+A `policy-server` receives the Admission Request object and,
+based on the endpoint that received the request,
+uses the correct policy to evaluate it.
 
-Each policy is evaluated inside of its own dedicated WebAssembly sandbox.
-The communication between `policy-server` (the "host") and the WebAssembly
-policy (the "guest") is done using the waPC communication protocol. This is
-covered in depth inside of [this](/writing-policies/index.md)
-section of the documentation.
+Each policy evaluates inside its own dedicated WebAssembly sandbox.
```

Comment:
I would prefer to leave the passive here, as they are xecuted by the policy-server, and don't evaluate themselves.

Reply:
ok, done.